ctags/main/parse.c

```diff
@@ -2229,13 +2229,20 @@ extern bool runParserInNarrowedInputStream (const langType language,
 					       unsigned long endLine, int endCharOffset,
 					       unsigned long sourceLineOffset)
 {
-	bool tagFileResized;
+	bool tagFileResized = false;
 	pushNarrowedInputStream (language,
 				 startLine, startCharOffset,
 				 endLine, endCharOffset,
 				 sourceLineOffset);
 #ifndef CTAGS_LIB
 	tagFileResized = createTagsWithFallback1 (language);
+#else
+	/* Simple parsing without rescans - not used by any sub-parsers anyway */
```

Comment:
OK, good point. I wanted to avoid re-parsing within subparsers because this would complicate things in the tag manager - if re-parsing happens in the main parser, we clear the tag list and start over. However, if such a thing happened within a subparser, we'd have to have some way to clear just the part of the tags generated by the subparser while leaving the tags generated by the main parser and we currently don't have anything like that in place.

Personally I'd just sacrifice re-parsing in this case (i.e. when using flex running cpp parser as the subparser) to simplify things. If I remember correctly, re-parsing happens quite rarely, at least when I tried the linux kernel sources.